### PR TITLE
Andrey Milko, MVP

### PR DESCRIPTION
**Pull request by Andrewregrets**
- Implemented weather info fetching by user's geolocation.
- Implemented weather info fetching of a constant set of cities.

### CODE REVIEW

File: src/components/svg/HomeSvg.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+export default function HomeSvg() {
```
Comment (AV29) [inline src/components/svg/HomeSvg.jsx]: Nice move ! :) 

Optionally you could use webpack loader instead importing svg's the same way later as you do with this approach. We can talk about it separately.

File: src/services/weatherService.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { getDayOfTheWeekName, getDateTime } from '../utils/dateUtils';
+import { API_KEY } from '../secrets';
+import { kelvinToCelsius } from '../utils/weatherUtils';
+
+const convertToCelsiusAndFloor = (t) => Math.floor(kelvinToCelsius(t));
```
Comment (AV29) [inline src/services/weatherService.js]: How about omitting parentheses if there's only one param ? ;)

File: src/services/weatherService.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        temperature: convertToCelsiusAndFloor(weather.main.temp),
+      }))
+    );
+
+export const getWeatherByGeolocation = ({ coords: { latitude, longitude } }) =>
+  fetch(
+    `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&APPID=${API_KEY}`
+  )
+    .then((resp) => resp.json())
+    .then(async (weather) => {
```
Comment (AV29) [inline src/services/weatherService.js]: Just curious: why did you decide to mix async/await with promises ?

File: src/services/weatherService.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { getDayOfTheWeekName, getDateTime } from '../utils/dateUtils';
```
Comment (AV29) [inline src/services/weatherService.js]: I like how you created service layer! Cool!

File: src/utils/dateUtils.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  'Sunday',
+  'Monday',
+  'Tuesday',
+  'Wednesday',
+  'Thursday',
+  'Friday',
+  'Saturday',
+];
+
+const makeGetDayOfTheWeek = (weekDayNames) => (date) => weekDayNames[date.getDay()];
```
Comment (AV29) [inline src/utils/dateUtils.js]: Nice!

But as **weekDayNamesEng** is a constant you could probably skip additional closure and read value directly like: 
`const getDayOfTheWeek = date => weekDayNamesEng[date.getDay()];`

What do you think ? Am I missing something ? Probably you wanted to reuse **makeGetDayOfTheWeek** somewhere else ?

File: src/utils/weatherUtils.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export const kelvinToCelsius = (kelvin) => kelvin - 273.15;
```
Comment (AV29) [inline src/utils/weatherUtils.js]: OPTIONAL: By convention you can totally omit parentheses if there is only one parameter

File: src/components/App.jsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    super(props);
+    this.state = {
+      currentPage: pages.currentWeather,
+    };
+  }
+
+  render() {
+    const { currentPage } = this.state;
+
+    const page =
```
[the remark this answers is not in the document]
Comment (Andrewregrets) [reply src/components/App.jsx]: Yep, I'll put it in the render return statement.

File: src/components/App.jsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            >
+              Back
+            </Button>
+          }
+        />
+      );
+    return (
+      <div style={appContainerStyle}>
+        <SideMenu
+          onHomeClick={() => this.setState({ currentPage: pages.currentWeather })}
```
[the remark this answers is not in the document]
Comment (Andrewregrets) [reply src/components/App.jsx]: Yes, I agree. I considered this, but decided to leave it for now. I think I should pass the array of objects with onNavItemClicked and svgComponent keys.

File: .prettierrc [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+  "printWidth": 100,
+  "singleQuote": true,
```
Comment (ximet) [inline .prettierrc]: Excellent setup!

File: src/components/CityList/CityListItem.jsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function CityListItem({ cityName, temperature }) {
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (Andrewregrets) [reply src/components/CityList/CityListItem.jsx]: What exactly? React.Fragment?

File: src/components/CityList/CityList.jsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+      <div
+        style={{
+          display: 'grid',
+          gridTemplateColumns: '1fr auto',
+          fontSize: '1.5rem',
+          gridGap: '1rem 2rem',
+        }}
+      >
+        {cities.map(({ name, temperature }) => (
+          <CityListItem key={name} cityName={name} temperature={temperature} />
```
[the remark this answers is not in the document]
Comment (ximet) [reply src/components/CityList/CityList.jsx]: agree with you

File: src/components/CityList/CityListItem.jsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (ximet) [reply src/components/CityList/CityListItem.jsx]: I like more informative code: React.Fragment

File: src/services/weatherService.js (hunk 2 above)
Comment (Andrewregrets) [reply src/services/weatherService.js]: Agree. I should also reconfigure the linter.

File: src/services/weatherService.js (hunk 3 above)
Comment (Andrewregrets) [reply src/services/weatherService.js]: After I receive `weather ` info from api.openweathermap.org I want to get full `country.name`. How would you do that?

**Pull request closed without merging**